### ISSUE TITLE
expandRow having no effect

### ISSUE THREAD

**Issue by christanfoden** [react-bootstrap-table/react-bootstrap-table2#460]
I seem to be having some difficulty using expandRow, I've followed the docs along and search for similar problems but can't find much. I don't get any errors but if I click on a row nothing happens. I could be making a basic error somewhere? 

here is my table component and i've tried using an alert inside the expandRow prop which fire on component mounting but for the life of me cannot why it's not working.

`<BootstrapTable
                      expandRow={expandRow}
                      hover
                      bordered={false}
                      striped
                      condensed
                      responsive
                      size="sm"
                      keyField="email"
                      data={this.props.subscribers}
                      columns={columns}
                      defaultSorted={defaultSorted}
                      filter={filterFactory()}
                      pagination={paginationFactory(options)}
                      noDataIndication="Loading ..."
                      cellEdit={cellEditFactory({
                        mode: "dbclick",
                        blurToSave: true,
                        afterSaveCell: (oldValue, newValue, row, column) =>
                          this.handleTypeChange(oldValue, newValue, row, column)
                      })}
                    />`

`const expandRow = {
      renderer: row => (
        <div>
          <p>{`This Expand row is belong to rowKey ${row.id}`}</p>
          <p>
            You can render anything here, also you can add additional data on
            every row object
          </p>
          <p>
            expandRow.renderer callback will pass the origin row object to you
          </p>
        </div>
      )
    };`

**Comment by blac3kman** [comment]
I use  "react-bootstrap-table-next": "1.0.0". When I click on expandedRow I receive 

```javascript
Warning: Failed prop type: Invalid prop `isAnyExpands` of type `boolean` supplied to `BootstrapTable`, expected `function`.
    in BootstrapTable
    in RowExpandProvider
    in DataProvider (created by BootstrapTableContainer)
    in BootstrapTableContainer (created by DataTableComponent)
```

**Comment by christanfoden** [comment]
hmm.... looks like you have got further than me, I must be using the wrong version I am on `    "react-bootstrap-table-next": "^0.1.15",
`
are you passing 'isAnyExpands' in as a prop to your `<BootstrapTable />` component?

**Comment by christanfoden** [comment]
Ok so I have upgraded to 1.0.0 which makes sense as the only one that works with Bootstrap 4, which I am running...using these versions:

`"react-bootstrap-table-next": "^1.0.0",
    "react-bootstrap-table2-editor": "^0.2.1",
    "react-bootstrap-table2-filter": "^0.3.0",
    "react-bootstrap-table2-paginator": "^0.1.6",
    "react-bootstrap-table2-toolkit": "^1.0.0",`

A bunch of errors were thrown up around 'filters' so I made a new table with none of the other dependencies involved (above ones) and it works!!! very cool!! Now i'll re implement one by one each dependency and ensure their version is compatible. I'll post the winning combination.

**Comment by blac3kman** [comment]
thanks. I'll wait your post!!!!

**Comment by christanfoden** [comment]
I'm not sure it works with `react-bootstrap-table2-filter` at all yet, just tried 1.0.0 and it doesn't seem even without being called by a column.

**Comment by christanfoden** [comment]
@AllenFang do you know when this is likely to happen? Assume its either being worked on, on the road map or I've overlooked something...

**Comment by blac3kman** [comment]
In my case I don't use any filter but I use pagination: 1.0.0. When I click on row everything is works but I can see in web console error message which I posted before.

`
Warning: Failed prop type: Invalid prop `isAnyExpands` of type `boolean` supplied to `BootstrapTable`, expected `function`.
    in BootstrapTable
    in RowExpandProvider
    in DataProvider (created by BootstrapTableContainer)
    in BootstrapTableContainer (created by DataTableComponent)
`

**Comment by AllenFang** [comment]
@blac3kman ok, it's probably a little bug that I can fix recently

@christanfoden Can you confirm you already upgrade all the packages of `react-bootstrap-table2` to newest(`1.0.0`)?  

Sorry for lately reply

**Comment by AllenFang** [comment]
@blac3kman BTW, if you don't use the pagination, the warning still come out?

**Comment by christanfoden** [comment]
Hey @AllenFang no worries, Yes I upgraded to 1.0.0 and if i comment out all the plugins it works well. But I then tried to re-introduce filters and it didn't work. So I upgraded filters to 1.0.0 which didn't work either. I then tried to go back to the verisons I originally had and got in all kinds of trouble so re instantiated my branch to before I tried updrading to 1.0.0

**Comment by AllenFang** [comment]
@christanfoden 👍  So any issue there? Filter work fine?

**Comment by blac3kman** [comment]
@AllenFang 
> BTW, if you don't use the pagination, the warning still come out?

It doesn't matter with pagination or not. Expandend mechanism works but with some error in console.

**Comment by christanfoden** [comment]
@Allenfang no it doesn’t work with filters, it crashes. I can try and replicate in a repo and post it here if that’s helpful?

**Comment by AllenFang** [comment]
@blac3kman Thank, it's minor issue because just a warning, I will check it out if I have time
@christanfoden sure, please give me a repo or code to reproduce, but keep it minimal and simple.
I didn't check expand row with bootstrap4, I will check it out.  Thanks

**Comment by AllenFang** [comment]
@blac3kman the warning was fixed in `react-bootstrap-table-next@1.1.0`

[RELEASE NOTE](https://react-bootstrap-table.github.io/react-bootstrap-table2/blog/2018/08/12/version-bump.html)

**Comment by blac3kman** [comment]
@AllenFang thank you very much))
